libraries/app/database_api.cpp

```diff
@@ -1526,7 +1526,7 @@ vector<discussion> database_api::get_discussions_by_cashout( const discussion_qu
       const auto& tidx = my->_db.get_index<tags::tag_index>().indices().get<tags::by_cashout>();
       auto tidx_itr = tidx.lower_bound( boost::make_tuple( tag, fc::time_point::now() - fc::minutes(60) ) );
 
-      return get_discussions( query, tag, parent, tidx, tidx_itr, query.truncate_body, []( const comment_api_obj& c ){ return c.children_rshares2 <= 0; } );
+      return get_discussions( query, tag, parent, tidx, tidx_itr, query.truncate_body, []( const comment_api_obj& c ) { return c.cashout_time <= c.created;  });
```

Comment:
The lambda argument is a filter. If the filter lambda returns true then the comment is not returned. The previous filter was to remove all comments whose discussion had more negative rshares.

I would pass in `filter_default` for the filter arg and then make the exit arg `[]( const comment_api_obj& c ){ return c.net_rshares <= 0; }` to stop returning results on negative payout posts.